File: app/src/main/java/com/leon/patmore/life/efficiency/client/LifeEfficiencyClient.java
package com.leon.patmore.life.efficiency.client;

import com.leon.patmore.life.efficiency.client.domain.ListItem;

import java.util.List;

public interface LifeEfficiencyClient {

    List<ListItem> getListItems() throws LifeEfficiencyException;

    List<String> getTodayItems() throws LifeEfficiencyException;

    void acceptTodayItems() throws LifeEfficiencyException;

    void addPurchase(String name, int quantity) throws LifeEfficiencyException;

    void addToList(String name, int quantity) throws LifeEfficiencyException;

    void completeItems(List<String> items) throws LifeEfficiencyException;

    List<String> getRepeatingItems() throws LifeEfficiencyException;

    void addRepeatingItem(String item) throws LifeEfficiencyException;

    void deleteListItem(String name, int quantity) throws LifeEfficiencyException;

    void completeItem(String name, int quantity) throws LifeEfficiencyException;

}

File: app/src/main/kotlin/com/leon/patmore/life/efficiency/client/LifeEfficiencyClient.kt
package com.leon.patmore.life.efficiency.client;

import com.leon.patmore.life.efficiency.client.domain.Goal
import com.leon.patmore.life.efficiency.client.domain.HistoryItem
import com.leon.patmore.life.efficiency.client.domain.ListItem
import com.leon.patmore.life.efficiency.client.domain.RepeatingItem
import com.leon.patmore.life.efficiency.client.domain.TodoItem
import com.leon.patmore.life.efficiency.client.domain.WeeklyItem

interface LifeEfficiencyClient {

    fun getListItems(): List<ListItem>

    fun getTodayItems(): List<String>

    fun addPurchase(name: String, quantity: Int)

    fun addToList(name: String, quantity: Int)

    fun completeItems(items: List<String>)

    fun getRepeatingItems(): List<String>

    fun getRepeatingItemsDetails(): Map<String, RepeatingItem>

    fun addRepeatingItem(item: String)

    fun deleteListItem(name: String, quantity: Int)

    fun completeItem(name: String, quantity: Int)

    fun getHistory() : List<HistoryItem>

    fun todoNonCompleted() : List<TodoItem>

    fun todoList(status: String? = null, sorted: Boolean = false) : List<TodoItem>

    fun updateTodoItemStatus(id: Int, status: String)

    fun addTodo(desc: String)

    fun getWeekly(): List<WeeklyItem>

    fun completeWeeklyItem(id: Int)

    fun getGoals(): Map<String, Map<String, List<Goal>>>

}
